### PR TITLE
Added a trustee enum, and added the use for the role method from the …

### PR DESCRIPTION
**Pull request by ChristofferAAU**
…api_client, allowing us to check what role a user has.

# Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes #A subtask of 783

## Type of change

*Please delete options that are not relevant.*

- [Check] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration*

- [Used a print string to verify that the method returned the correct role] Test A
- [Unit test that checks if the mode is set to trustee when setting it to trustee] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [Check] My code follows the style guidelines of this project
- [Check] I have performed a self-review of my own code
- [Check] I have commented my code, particularly in hard-to-understand areas
- [Check] I have made corresponding changes to the documentation
- [Check] My changes generate no new warnings
- [Check] I have added tests that prove my fix is effective or that my feature works
- [Check] New and existing unit tests pass locally with my changes
- [Check] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by AdrianPlesner** [comment]
Why do request me as reviewer?

**Comment by JakobFaarGreg** [comment]
- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code does not introduce functionality that is unnecessary for solving the problem

Code Readability 

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no "hack", or obscure solution

Code Maintainability

- [x] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does not need to be covered by tests
- [x] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [x] All functionality is covered by tests
- [x] The tests has sensible names
- [x] By reading the tests you know what they cover
- [x] The tests cover sensible cases. Both happy paths and exception paths
- [x] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

Functionality

For this part, run the web-api locally as described in https://aau-giraf.github.io/wiki/development/rest_api_development/BuildAndRunLocally/ 
- [x] The web-api runs without issues
- [x] The functionality that the code claims to implement, are actually implemented
- [ ] The integration tests runs successfully. Refer to https://github.com/aau-giraf/web-api/tree/master/GirafIntegrationTest for how to run the tests

File: test/blocs/auth_bloc.test.dart [hunk 1]
```diff
@@ -38,4 +38,17 @@ void main() {
...
+      'when setMode is called with trustee'
+      ,
+      async((DoneFn done) {
+        authBloc.mode.skip(1).listen((WeekplanMode mode) {
+          expect(mode, WeekplanMode.trustee);
+          done();
+        });
+        authBloc.setMode(WeekplanMode.trustee);
+      })
+  );
```
Comment (JakobFaarGreg) [inline test/blocs/auth_bloc.test.dart:77]: Hey hey, den her test er ikke som sådan en test af den kode du har skrevet i `auth_bloc.dart`.
Du bør i stedet teste metoden authenticate som det her ligger i.
Den test bør handle om, om du kan ende inde i de kontrol strukturerer du har tilføjet til metoden.

**Review by JakobFaarGreg — COMMENTED** [review]
Added comment to code

**Review by kristnaKris — COMMENTED** [review]
Udover at testene skal udvides til at dække hele authenticate metoden i forhold til if else kæden med user type som Jakob har kommenteret så kan jeg approve jeres kode

**Comment by JakobFaarGreg** [comment]
> Udover at testene skal udvides til at dække hele authenticate metoden i forhold til if else kæden med user type som Jakob har kommenteret så kan jeg approve jeres kode

+1

File: test/blocs/auth_bloc.test.dart (hunk 1 above)
Comment (ChristofferAAU) [reply test/blocs/auth_bloc.test.dart:77]: Der burde være test på hele authenticate nu, i tilfælde af at de er citizen, trustee, eller guardian.

File: test/blocs/auth_bloc.test.dart [hunk 2]
```diff
@@ -51,4 +81,35 @@ void main() {
         authBloc.setMode(WeekplanMode.trustee);
       })
   );
+
+  const String username = 'Graatand';
+  const String password = 'password';
+  test('Should check that authenticate works', async((DoneFn done) {
+    authBloc.mode.skip(1).listen((WeekplanMode mode) {
+
```
Comment (JakobFaarGreg) [inline test/blocs/auth_bloc.test.dart]:
```suggestion
```

File: test/blocs/auth_bloc.test.dart [hunk 3]
```diff
@@ -1,15 +1,45 @@
...
+    } else if (username.compareTo('Chris') == 0){
+
+      return Stream<String>.value('Trustee');
+
+    } else if (username.compareTo('Janne') == 0){
+
+      return Stream<String>.value('Citizen');
+    }
+
```
Comment (JakobFaarGreg) [inline test/blocs/auth_bloc.test.dart]:
```suggestion
```

File: test/blocs/auth_bloc.test.dart [hunk 4]
```diff
@@ -1,15 +1,45 @@
...
+    if (username.compareTo('Graatand') == 0){
+
+      return Stream<String>.value('Guardian');
+
+    } else if (username.compareTo('Chris') == 0){
+
+      return Stream<String>.value('Trustee');
+
+    } else if (username.compareTo('Janne') == 0){
+
```
Comment (JakobFaarGreg) [inline test/blocs/auth_bloc.test.dart]:
```suggestion
```

File: test/blocs/auth_bloc.test.dart [hunk 5]
```diff
@@ -1,15 +1,45 @@
...
+  }
+
+  @override Stream<String> role(String username){
+    ///Returns a role to check that authenticate works
+    if (username.compareTo('Graatand') == 0){
+
+      return Stream<String>.value('Guardian');
+
+    } else if (username.compareTo('Chris') == 0){
+
```
Comment (JakobFaarGreg) [inline test/blocs/auth_bloc.test.dart]:
```suggestion
```

File: test/blocs/auth_bloc.test.dart [hunk 6]
```diff
@@ -1,15 +1,45 @@
...
+    ///Returns true to mark the the user exists
+   return Stream<bool>.value(true);
+  }
+
+  @override Stream<String> role(String username){
+    ///Returns a role to check that authenticate works
+    if (username.compareTo('Graatand') == 0){
+
+      return Stream<String>.value('Guardian');
+
```
Comment (JakobFaarGreg) [inline test/blocs/auth_bloc.test.dart]:
```suggestion
```

File: test/blocs/auth_bloc.test.dart [hunk 7]
```diff
@@ -1,15 +1,45 @@
...
+class MockAccountApi extends Mock implements AccountApi {
+  @override Stream<bool> login(String username, String password){
+    ///Returns true to mark the the user exists
+   return Stream<bool>.value(true);
+  }
+
+  @override Stream<String> role(String username){
+    ///Returns a role to check that authenticate works
+    if (username.compareTo('Graatand') == 0){
+
```
Comment (JakobFaarGreg) [inline test/blocs/auth_bloc.test.dart]:
```suggestion
```

File: test/blocs/auth_bloc.test.dart [hunk 8]
```diff
@@ -1,15 +1,45 @@
...
+      return Stream<String>.value('Guardian');
+
+    } else if (username.compareTo('Chris') == 0){
+
+      return Stream<String>.value('Trustee');
+
```
Comment (JakobFaarGreg) [inline test/blocs/auth_bloc.test.dart]:
```suggestion
```

File: test/blocs/auth_bloc.test.dart [hunk 9]
```diff
@@ -1,15 +1,45 @@
...
 void main() {
-  Api api;
+  Api _api;
   AuthBloc authBloc;
   setUp((){
-    api = Api('any');
-    authBloc = AuthBloc(api);
+    _api = Api('any');
+    authBloc = AuthBloc(_api);
+  _api.account = MockAccountApi();
```
Comment (JakobFaarGreg) [inline test/blocs/auth_bloc.test.dart]:
```suggestion
    _api.account = MockAccountApi();
```

File: test/blocs/auth_bloc.test.dart [hunk 10]
```diff
@@ -51,4 +81,35 @@ void main() {
...
+      expect(mode, WeekplanMode.guardian);
+      done();
+    });
+    authBloc.authenticate(username, password);
+  }));
+
+  const String username2 = 'Chris';
+  test('Should check that authenticate works', async((DoneFn done) {
+    authBloc.mode.skip(1).listen((WeekplanMode mode) {
+
```
Comment (JakobFaarGreg) [inline test/blocs/auth_bloc.test.dart]:
```suggestion
```

File: test/blocs/auth_bloc.test.dart [hunk 11]
```diff
@@ -51,4 +81,35 @@ void main() {
...
+      expect(mode, WeekplanMode.trustee);
+      done();
+    });
+    authBloc.authenticate(username2, password);
+  }));
+
+  const String username3 = 'Janne';
+  test('Should check that authenticate works', async((DoneFn done) {
+    authBloc.mode.skip(1).listen((WeekplanMode mode) {
+
```
Comment (JakobFaarGreg) [inline test/blocs/auth_bloc.test.dart]:
```suggestion
```

**Review by JakobFaarGreg — COMMENTED** [review]
Der er lige noget whitespace der skal elimineres

**Review by ChristofferAAU — COMMENTED** [review]
Check

**Review by JakobFaarGreg — APPROVED** [review]
Nu er jeg klar til at approve ;)

**Review by kristnaKris — APPROVED** [review]

**Pull request merged**